### PR TITLE
Adds resin gargoyles

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
25 tac points, alerts you of enemies within 10 tiles of it

## Why It's Good For The Game

Muh structure content

## Changelog
:cl:
add: Added resin gargoyles, a xeno structure that alerts xenos to nearby enemies
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]
lmao that sprite

**Comment by TiviPlus** [comment]
I had to color the red one myself im not proud

**Comment by TiviPlus** [comment]
sprite was from cm btw

File: code/modules/xenomorph/resin_gargoyle.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+/obj/structure/xeno/resin_gargoyle/proc/clear_warning()
+	is_alerting = FALSE
+	update_minimap_icon()
+	update_appearance()
+
+/obj/structure/xeno/resin_gargoyle/update_icon_state()
+	. = ..()
+	icon_state = is_alerting ? "gargoyle_alarm" : "gargoyle"
+
+/obj/structure/xeno/resin_gargoyle/proc/update_minimap_icon()
```
Comment (ivanmixo) [inline code/modules/xenomorph/resin_gargoyle.dm:57]: doc

File: code/modules/xenomorph/resin_gargoyle.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/obj/structure/xeno/resin_gargoyle
+	name = "resin gargoyle"
+	desc = "A resin monument to your tresspass. Alerts the xenomorph hive when an enemy approaches."
+	icon = 'icons/Xeno/2x2building.dmi'
+	icon_state = "gargoyle"
+	max_integrity = 100
+	///Bool if we're currently alerting
+	var/is_alerting = FALSE
+	COOLDOWN_DECLARE(proxy_alert_cooldown)
```
Comment (ivanmixo) [inline code/modules/xenomorph/resin_gargoyle.dm:9]: should doc the cooldown too even if it is self evident

**Review by ivanmixo — CHANGES_REQUESTED** [review]
Wonder if this'll be worth using over just a sticky turret, I guess it is half the price

**Review by ivanmixo — APPROVED** [review]

File: code/modules/xenomorph/resin_gargoyle.dm [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	max_integrity = 100
+	///Bool if we're currently alerting
+	var/is_alerting = FALSE
+	//cd tracking for the alert
+	COOLDOWN_DECLARE(proxy_alert_cooldown)
+
+/obj/structure/xeno/resin_gargoyle/Initialize(mapload, _hivenumber, mob/living/carbon/xenomorph/creator)
+	. = ..()
+	for(var/turfs in RANGE_TURFS(XENO_GARGOYLE_DETECTION_RANGE, src))
+		RegisterSignal(turfs, COMSIG_ATOM_ENTERED, PROC_REF(gargoyle_alarm))
```
Comment (DeltaFire15) [inline code/modules/xenomorph/resin_gargoyle.dm:15]: Should unregister these signals on destroy

Comment (ivanmixo) [reply code/modules/xenomorph/resin_gargoyle.dm:15]: https://github.com/tgstation/TerraGov-Marine-Corps/blob/e0a7b011bc5def4075ef345019d14b1e1a71b4e9/code/datums/datum.dm#L129

Shouldn't be necesarry

Comment (Lumipharon) [reply code/modules/xenomorph/resin_gargoyle.dm:15]: yeah signals are all unregistered on destroy already.

**Comment by Ragecrabbo** [comment]
Does this trigger on line of sight like a turret or whenever anything comes within close proximity, regardless of obstructions? 
Either way i think this is a great addition as a sort of cheap doorbell for places xenos want to defend like disks, cave entrances, SD or even in middle of a bunch of recently dead marines.

**Pull request merged**